app/models/metric/ci_mixin/processing.rb

```diff
@@ -197,4 +199,23 @@ def normalize_value(value, counter)
     end
     return value, message
   end
+
+  def publish_metrics(metrics)
+    return if MiqQueue.messaging_type == "miq_queue"
+
+    metrics.each_value do |metric|
+      resource = metric.delete(:resource)
+      metric[:resource_type] = resource.class.base_class.name
+      metric[:resource_id]   = resource.id
+      metric[:resource_ref]  = resource.ems_ref if resource.respond_to?(:ems_ref)
```

Comment:
Note sure on the name `resource_ref`... thinking it should say manager or native or something in there (resource_native_ref or resource_manager_ref)

Reply:
:+1: done